webnn/clamp.https.any.js

```diff
@@ -0,0 +1,214 @@
+// META: title=test WebNN API clamp operation
+// META: global=window,dedicatedworker
+// META: script=./resources/utils.js
+// META: timeout=long
+
+'use strict';
+
+// https://webmachinelearning.github.io/webnn/#api-mlgraphbuilder-clamp
+
+const testClamp = async (operandType, syncFlag, inputValue, inputShape, expected, options) => {
+  const x = builder.input('x', {type: operandType, dimensions: inputShape});
+  const y = builder.clamp(x, options);
+  const TestTypedArray = TypedArrayDict[operandType];
+  const inputs = {'x': new TestTypedArray(inputValue)};
+  const outputs = {'y': new TestTypedArray(sizeOfShape(inputShape))};
+  let graph;
+
+  if (syncFlag) {
+    graph = builder.build({y});
+    context.compute(graph, inputs, outputs);
+  } else {
+    graph = await builder.buildAsync({y});
+    await context.computeAsync(graph, inputs, outputs);
+  }
+
+  const expectedData = expected instanceof Array ? new TestTypedArray(expected) : expected[operandType];
+  assert_array_approx_equals_ulp(outputs.y, expectedData, PrecisionMetrics.ULP[operandType].clamp, operandType);
+};
+
+// Input data are random float64 numbers between [-10, 10) generated by invoking getRandomArbitrary(-10, 10)
+// refering to https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random#getting_a_random_number_between_two_values
+const inputData = [
+  -3.4356449350874696, -6.530945988411405,  -8.175760663838268, 2.0879641317522726,
+  -4.480150236948526,  -8.591504561715722,  5.071455429211573,  -6.618697702258771,
+  4.224577823136105,   6.450272349350044,   -8.799923845835664, -3.3445965406946643,
+  5.550524270215341,   1.2788677438688012,  9.333702625514768,  9.2261637863086,
+  -7.302720212371034,  1.7865902395032585,  5.564981581526375,  3.145101011211482,
+  -8.275078596251655,  -1.3557080837143296, 7.348269585030259,  -5.530012756488021,
+];
+// expected data by clamping input data with default options
+const expectedDefault = {
```

Comment:
Could you please share how to get those expected data by running https://github.com/webmachinelearning/webnn-baseline against the `inputData`? And probably share the script that others can reproduce the expected results.

Reply:
It's also important (even more so) that the input be typed than the output. e.g.

```
const inputData = [
    float32: [
        -3.4356448650360107421875, -6.53094577789306640625, -8.1757602691650390625, 2.08796405792236328125, 
        -4.4801502227783203125, -8.59150409698486328125, 5.071455478668212890625, -6.618697643280029296875, 
        4.22457790374755859375, 6.45027256011962890625, -8.79992389678955078125, -3.3445966243743896484375, 
        5.550524234771728515625, 1.2788677215576171875, 9.33370304107666015625, 9.2261638641357421875, 
        -7.30272006988525390625, 1.78659021854400634765625, 5.5649814605712890625, 3.145101070404052734375, 
        -8.27507877349853515625, -1.35570812225341796875, 7.34826946258544921875, -5.530012607574462890625, 
    ]
    float64: [
        -3.4356449350874696, -6.530945988411405,  -8.175760663838268, 2.0879641317522726,
        -4.480150236948526,  -8.591504561715722,  5.071455429211573,  -6.618697702258771,
        4.224577823136105,   6.450272349350044,   -8.799923845835664, -3.3445965406946643,
        5.550524270215341,   1.2788677438688012,  9.333702625514768,  9.2261637863086,
        -7.302720212371034,  1.7865902395032585,  5.564981581526375,  3.145101011211482,
        -8.275078596251655,  -1.3557080837143296, 7.348269585030259,  -5.530012756488021,
    ]
];
```